pkg/query/storeset.go

```diff
@@ -420,7 +445,13 @@ func (s *StoreSet) getActiveStores(ctx context.Context, stores map[string]*store
 			st, seenAlready := stores[addr]
 			if !seenAlready {
 				// New store or was unactive and was removed in the past - create new one.
-				conn, err := grpc.DialContext(ctx, addr, s.dialOpts...)
+				tlsCfg, err := tls.NewClientConfig(s.logger, s.cert, s.key, s.caCert, spec.ServerName())
+				if err != nil {
+					level.Warn(s.logger).Log("msg", "update of store node failed", "err", errors.Wrap(err, "setting TLS"), "address", addr)
+					return
+				}
+				dialOpts := append(s.dialOpts, grpc.WithTransportCredentials(credentials.NewTLS(tlsCfg)))
+				conn, err := grpc.DialContext(ctx, addr, dialOpts...)
```

Comment:
I believe this makes an assumption that store is listening with TLS & won't work in insecure mode. Could you please verify that?

Reply:
Yes, That would be absolutely true.  I need to conditionally do this based on the combination of the secure flag and the new grpc-client-dns-server-name flag.